drivers/usb/host/dwc_otg/dwc_otg_hcd.c

```diff
@@ -1858,7 +1858,7 @@ int fiq_fsm_queue_split_transaction(dwc_otg_hcd_t *hcd, dwc_otg_qh_t *qh)
 	st->hctsiz_copy.b.pid = hc->data_pid_start;
 
 	if (hc->ep_is_in || (hc->xfer_len > hc->max_packet)) {
-		hc->xfer_len = hc->max_packet;
+		hc->xfer_len = min_t(uint32_t, hc->max_packet, hc->xfer_len);
```

Comment:
Swapping the order of parameters compared to L1816 causes me a minor discomfort... (although obviously it works the same)